test/unit/core/configuration/configuration_test.cpp

```diff
@@ -171,6 +171,72 @@ TEST(configuration, exists_by_type_template)
     EXPECT_FALSE(decltype(cfg)::exists<foo>());
 }
 
+TEST(configuration, append_configuration_element)
+{
+    {
+        seqan3::configuration<foo, bar> cfg{};
+        auto new_cfg = cfg.append(bax{});
+        EXPECT_TRUE((std::same_as<decltype(new_cfg), seqan3::configuration<foo, bar, bax>>));
+    }
+
+    {
+        seqan3::configuration<foo, bar> cfg{};
+        bax b{};
+        auto new_cfg = cfg.append(b);
+        EXPECT_TRUE((std::same_as<decltype(new_cfg), seqan3::configuration<foo, bar, bax>>));
+    }
+
+    {
+        seqan3::configuration<foo, bar> cfg{};
+        bax b{};
+        auto new_cfg = cfg.append(std::as_const(b));
+        EXPECT_TRUE((std::same_as<decltype(new_cfg), seqan3::configuration<foo, bar, bax>>));
+    }
+
+    {
+        seqan3::configuration<> cfg{};
+        bax b{};
+        auto new_cfg = cfg.append(std::as_const(b));
```

Comment:
Just for my self: lvalue const, but empty lhs

Hmm from the above scheme, I would have expected `rvalue const`

Reply:
This case I would expect in `append_configuration_element_special_cases`

Reply:
Yeah, but I am not really sure if `const &&` is a necessary use case. Simply, because the configuration itself is not really generic since the types (configuration elements) that are added are fixed.

Reply:
Okay